### PR TITLE
Add len for Slice[T] where T is ordinal (#5703)

### PR DESCRIPTION
**Pull request by cooldome**
I had to move len definition after +,- operator definitions.
Checked generated C code, no unexpected deficiencies

Let me know if tests are required

### CODE REVIEW

**Comment by cooldome** [comment]
@Araq could you please comment this please? Do you think Slice[T: ordinal] needs a len proc or not? if yes, let me know of you preferred approach

**Comment by Araq** [comment]
I don't really like it and the diff is much larger than necessary too.

**Comment by cooldome** [comment]
Could you be slightly more specific?  If you say Slice[T] should have len proc, but implemented in different way, give a me hint what you want and I'll do it. If you say Slice[T] doesn't need length then we can cancel.

Diff is large because I wanted all len procs to be together and as it currently stands +,- defined lower in system.nim so I had to move. If you prefer a small diff that is dead easy to fix, just one len will hang out at different location from the rest

**Comment by Araq** [comment]
>  If you prefer a small diff that is dead easy to fix, just one len will hang out at different location from the rest

That's what I meant, yes.

File: lib/system.nim [hunk 1]
```diff
@@ -1269,6 +1253,28 @@ proc add*(x: var string, y: string) {.magic: "AppendStrStr", noSideEffect.}
...
+  ## .. code-block:: nim
+  ##  var arr = [1,1,1,1,1]
+  ##  len(arr) #=> 5
+  ##  for i in 0..<arr.len:
+  ##    echo arr[i] #=> 1,1,1,1,1
+
+proc len*[T: Ordinal](x: Slice[T]): int {.inline, noSideEffect.} =
+  ## returns the length of an ordinal slice
+  result = ord(x.b) - ord(x.a) + 1
+  if result < 0: result = 0
```
Comment (dom96) [inline lib/system.nim]: Wouldn't it make more sense to use ``abs``?

Comment (cooldome) [inline lib/system.nim]: (for i in 5..2)  gives zero iterations not 4, imo len should be consistent with current behavior

Comment (cooldome) [inline lib/system.nim]: proc len*[T: Ordinal](x: Slice[T]): int {.noSideEffect, .inline.} =
  result = max(0, ord(x.b) - ord(x.a) + 1)

would be in the next pull request, given now i can put len after max definition

**Comment by cooldome** [comment]
I have submitted a new pull request. I closing this one, because I had to remove branch

**Pull request closed without merging**